### ISSUE TITLE
scaladoc can fail at specific macro annotations (Regression from scala 2.11.1 to 2.11.2)

### ISSUE THREAD

**Issue by FlorianKirmaier** [scalamacros/paradise#53]
I can compile the following code with "sbt compile" but "sbt doc" fails.
It works with scala 2.11.1 but fails with scala 2.11.2.
I tested paradise version "2.0.1" and "2.1.0-SNAPSHOT".

```
  import scala.reflect.macros._
  class TestDoc extends StaticAnnotation {
    def macroTransform(annottees: Any*): Any = macro testDocImpl
  }
  def testDocImpl(c: blackbox.Context)(annottees: c.Tree*): c.Tree = {
    import c._
    import c.universe._
    var cc = annottees(0)

    val q"class $clazz(...$args)" = cc
    q"class $clazz(a: Int)"   // this ast can be created
    q"class $clazz(...$args)" // this fails!
  }
```

```
      @TestDoc class B(x: Int)
```

```
[error] C:\Users\florian.kirmaier\Dropbox\git\SimpleFX\core\src\main\scala\simpl
efx\Test.scala:17: exception during macro expansion:
[error] java.lang.UnsupportedOperationException: Position.point on NoPosition
[error]         at scala.reflect.internal.util.Position.fail(Position.scala:17)
[error]         at scala.reflect.internal.util.UndefinedPosition.point(Position.
scala:95)
[error]         at scala.reflect.internal.util.UndefinedPosition.point(Position.
scala:90)
[error]         at scala.reflect.internal.Positions$class.wrappingPos(Positions.
scala:42)
[error]         at scala.reflect.internal.SymbolTable.wrappingPos(SymbolTable.sc
ala:16)
[error]         at scala.reflect.internal.Positions$class.wrappingPos(Positions.
scala:37)
[error]         at scala.reflect.internal.SymbolTable.wrappingPos(SymbolTable.sc
ala:16)
[error]         at scala.reflect.internal.TreeGen.mkTemplate(TreeGen.scala:378)
[error]         at scala.reflect.internal.ReificationSupport$ReificationSupportI
mpl$SyntacticClassDef$.apply(ReificationSupport.scala:341)
[error]         at scala.reflect.internal.ReificationSupport$ReificationSupportI
mpl$SyntacticClassDef$.apply(ReificationSupport.scala:326)
[error]         at simplefx.wrapping.javafx.package$.testDocImpl(package.scala:6
3)
[error]       @TestDoc class B(x: Int)
[error]        ^
```

**Comment by xeno-by** [comment]
Hmm that's not good. Is this a blocker for you?

**Comment by FlorianKirmaier** [comment]
It blocks me from upgrading to 2.11.2.
I can use 2.11.1 for the next 3 weeks.
After that, i can work around it, by doing a special 2.11.1 build for the docs.
So no, it doesn't block, but it's realy annoying.

**Comment by xeno-by** [comment]
I'll see what I can do in the coming days. Thank you for reporting!

**Comment by xeno-by** [comment]
This is a bug in quasiquotes: https://issues.scala-lang.org/browse/SI-8771. Please follow the discussion over there - I'll try to figure out what can be done as a workaround.

**Comment by xeno-by** [comment]
Hi @FlorianKirmaier! Today I faced a similar bug, but the workaround proposed in SI-8771 wasn't enough. After some thinking, I found a way to attack the issue at its root, and now it shouldn't be a problem anymore, regardless of whether the workaround is used or now. The fix has been published under version 2.0.2-SNAPSHOT, 2.1.0-SNAPSHOT and 2.1.0-M3.

**Comment by FlorianKirmaier** [comment]
Great to see progress here!

 The original code of the sample works now. But unfortunately, I still have a lot of issues with `sbt doc`. I still get many `java.lang.UnsupportedOperationException: Position.point on NoPosition` errors.

Another error can be reproduced with the identity annotation from http://docs.scala-lang.org/overviews/macros/annotations.html
`
/** any doc */
@identity class A
`
